Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabAdapter.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <AzToolsFramework/Prefab/DocumentPropertyEditor/PrefabAdapter.h>
+#include <AzToolsFramework/Prefab/Overrides/PrefabOverridePublicInterface.h>
+
+namespace AzToolsFramework::Prefab
+{
+    PrefabAdapter::PrefabAdapter()
+    {
+        m_prefabOverridePublicInterface = AZ::Interface<AzToolsFramework::Prefab::PrefabOverridePublicInterface>::Get();
```

Comment:
Unless the PrefabOverridPublicPublicInterface is being held by reference, I don't see a good reason to hold onto it as a member.

It is dangerous if the registered PrefabOverridPublicPublicInterface needs to get replace on the AZ::Interface during the run.

Reply:
PrefabOverridePublicHandler is a singleton that is initialized once when the editor starts. It registers with the PrefabOverridPublicPublicInterface when it is constructed and doesn't unregister from it until the application is terminated. So we will not run into the issue you mentioned above.

I agree that holding onto raw pointers is always risky. But it has been common practice in the prefab code to hold on to interface pointers backed by singletons to avoid fetching them repeatedly and making the code bloated (because interfaces are used heavily in prefab code). So I am inclined to keep it this way for now.